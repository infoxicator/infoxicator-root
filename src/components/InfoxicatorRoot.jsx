import React, { Fragment } from 'react';
import ModuleRoute from 'holocron-module-route';
import Helmet from 'react-helmet';

const InfoxicatorRoot = ({ children }) => (
  <Fragment>
    <Helmet
      title="Infoxicator | One App | Microfrontend | tutorial | React"
      link={[
        { rel: 'stylesheet', href: 'https://cdn.jsdelivr.net/npm/bulma@0.8.0/css/bulma.min.css' },
      ]}
    />
    {children}
  </Fragment>
);

// Read about childRoutes:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/Routing.md#childroutes
InfoxicatorRoot.childRoutes = () => [
  <ModuleRoute path="/" moduleName="infoxicator-main" />,
];

// Read about appConfig:
// https://github.com/americanexpress/one-app/blob/master/docs/api/modules/App-Configuration.md
if (!global.BROWSER) {
  // eslint-disable-next-line global-require
  InfoxicatorRoot.appConfig = require('../appConfig').default;
}

export default InfoxicatorRoot;
