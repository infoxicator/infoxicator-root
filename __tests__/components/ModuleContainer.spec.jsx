import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AppConfig from '../../src/appConfig';
import InfoxicatorRoot from '../../src/components/InfoxicatorRoot';

describe('InfoxicatorRoot', () => {
  it('default export should return a function', () => {
    expect(InfoxicatorRoot).toBeInstanceOf(Function);
  });

  it('module should render correct JSX', () => {
    const renderedModule = shallow(<InfoxicatorRoot />);
    expect(toJson(renderedModule)).toMatchSnapshot();
  });

  // test only necessary for root modules
  it('appConfig should contain accurate csp', () => {
    expect(AppConfig.csp).toBeDefined();
    expect(typeof AppConfig.csp).toBe('string');
  });
});
