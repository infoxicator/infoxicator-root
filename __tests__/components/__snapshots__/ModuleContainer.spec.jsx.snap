// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoxicatorRoot module should render correct JSX 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    link={
      Array [
        Object {
          "href": "https://maxcdn.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css",
          "rel": "stylesheet",
        },
        Object {
          "href": "https://cdnjs.cloudflare.com/ajax/libs/prism/1.19.0/themes/prism-tomorrow.min.css",
          "rel": "stylesheet",
        },
        Object {
          "href": "https://cdnjs.cloudflare.com/ajax/libs/prism/1.19.0/plugins/line-numbers/prism-line-numbers.min.css",
          "rel": "stylesheet",
        },
        Object {
          "href": "https://www.infoxication.net/wp-content/uploads/2020/03/favicon.jpg",
          "rel": "icon",
          "sizes": "16x16",
          "type": "image/jpg",
        },
      ]
    }
    meta={
      Array [
        Object {
          "content": "width=device-width, initial-scale=1",
          "name": "viewport",
        },
      ]
    }
    title="Infoxicator | One App | Microfrontend | tutorial | React"
  />
  <ErrorBoundary>
    <Header />
    <Footer />
  </ErrorBoundary>
</Fragment>
`;
